core-bundle/src/EventListener/DataContainer/PageSearchListener.php

```diff
@@ -41,10 +41,10 @@ public function onSaveAlias(string $value, DataContainer $dc): string
         return $value;
     }
 
-    #[AsCallback(table: 'tl_page', target: 'fields.noSearch.save')]
-    public function onSaveNoSearch(string $value, DataContainer $dc): string
+    #[AsCallback(table: 'tl_page', target: 'fields.searchIndexer.save')]
+    public function onSaveSearchIndexer(string $value, DataContainer $dc): string
     {
-        if (!$value || (bool) $value === (bool) ($dc->getCurrentRecord()['noSearch'] ?? false)) {
+        if ($value === ($dc->getCurrentRecord()['searchIndexer'] ?? null) || !str_starts_with($value, 'never')) {
```

Comment:
I think this is still not correct. We want to purge the search index, if the value changes, right?

Reply:
- If `searchIndexer` was changed to the Blank option (= use robots tag), purging is not necessary. In this case the robots tag should trigger if the page should be added to the search index or if the index should be purged.
- If `searchIndexer` was changed to `always_index`, purging is not necessary. If the page wasn't in the search index, it should be added.
- If `searchIndexer` was changed to `never_index`, the search index should be purged.

I think the `if ()` condition is correct. What I still don't know is where the 'add to the index' action is triggered (as mentioned [here](https://github.com/contao/contao/pull/8252#issuecomment-2770737479) already)

Reply:
I think it needs to be:

```suggestion
        if (!$value || 'always_index' === $value || $value === ($dc->getCurrentRecord()['searchIndexer'] ?? null)) {
```

i.e. return if:

* The value is empty
* or if the value is `always_index`
* or if the value is the same as before.

Reply:
yeah that basically the same just inverted